src/annotation.ts

```diff
@@ -146,8 +149,9 @@ export class Annotation extends Modifier {
         annotation.setTextLine(state.text_line);
       }
     }
-    state.left_shift += width / 2;
-    state.right_shift += width / 2;
+    const overlap = Math.min(Math.max(width - maxGlyphWidth, 0), Math.max(width - state.left_shift * 2, 0));
```

Comment:
why is `state.right_shift` not part of the equation?

Reply:
fixed